pkg/controller/windowsmachineconfig/windowsmachineconfig_controller.go

```diff
@@ -166,6 +202,36 @@ func (r *ReconcileWindowsMachineConfig) getCloudProvider(instance *wmcapi.Window
 // Result.Requeue is true, otherwise upon completion it will remove the work from the queue.
 func (r *ReconcileWindowsMachineConfig) Reconcile(request reconcile.Request) (reconcile.Result, error) {
 	log.Info("reconciling", "namespace", request.Namespace, "name", request.Name)
+	wmcoObject := false
+	// Fetch the Machine instance
+	Machine := &mapiv1.Machine{}
+	provisionedPhase := "Provisioned"
+	if err := r.client.Get(context.TODO(), request.NamespacedName, Machine); err != nil {
+		wmcoObject = true
+		// TODO: Uncomment this when we switch to Machine-api completely
+		//if k8sapierrors.IsNotFound(err) {
+		//	// Request object not found, could have been deleted after reconcile request.
+		//	// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
+		//	// Return and don't requeue
+		//
+		//	return reconcile.Result{}, nil
+		//
+		//}
+		//// Error reading the object - requeue the request.
+		//return reconcile.Result{}, err
+	}
+	if !wmcoObject {
+		if Machine.Status.Phase == nil {
+			// Phase can be nil and should be ignored by WMCO
+			return reconcile.Result{}, nil
+		}
+		if *Machine.Status.Phase != provisionedPhase {
+			// If the Machine is not in provisioned state, WMCO shouldn't care about it
+			return reconcile.Result{}, nil
+		}
+		r.recorder.Eventf(Machine, corev1.EventTypeNormal, "WMCO Update",
```

Comment:
Not sure what `WMCO Update` implies here. Won't `WMCO` be sufficient?

Reply:
In testing, we are filtering the event using this string i.e `WMCO Update` and doing `string.contains` to check for events. 
We can use WMCO as well, in place of this. 
or we can reserve `WMCO` for events which state the `machine %s is provisioned as a node`.

Reply:
We can stick to the reason as `WMCO` for now and expand over time.

Reply:
It will be hard to parse and to filter:

For example 
-  `WMCO Update` when WMCO is `updating the machine to make it node`.
-  `WMCO Upgrade` when WMCO `issues a command for WMCB to stop kubelet for upgrade`
- `WMCO Failure` when WMCO `configuration has failed`

Reply:
Then `WMCO Update` can be confusing when there is `WMCO Upgrade`. How about `WMCO Setup`?